### ISSUE TITLE
[Bug]: MODEL grants producing repeated plan after apply

### ISSUE THREAD

**Issue by halcarrigan-carvana** [Snowflake-Labs/terraform-provider-snowflake#3050]
### Terraform CLI Version

1.9.2

### Terraform Provider Version

0.95.0

### Terraform Configuration

```terraform
resource "snowflake_role" "role_test_models" {
  name    = "ROLE_TEST_MODELS"
  provider = snowflake.useradmin
}

resource "snowflake_grant_account_role" "role_grant" {
  role_name        = snowflake_role.role_test_models.name
  parent_role_name = "SYSADMIN"
  provider = snowflake.securityadmin
}

resource "snowflake_grant_privileges_to_account_role" "grant_schema" {
  account_role_name = snowflake_role.role_test_models.name
  privileges         = ["USAGE"]
  with_grant_option  = false

  on_schema_object {
    future {
      object_type_plural = "MODELS"
      in_database        = "DB_TEST_MODELS"
    }
  }
  provider = snowflake.securityadmin
}
```

### Category

category:grants

### Object type(s)

resource:grant_privileges_to_account_role

### Expected Behavior

The above configuration creates a role, grants the role to SYSADMIN, and grants USAGE on FUTURE MODELS to that role. After applying USAGE on MODELS to role, subsequent plans should recommend no changes.

### Actual Behavior

Plan does not converge. Each run suggests adding the usage privilege again.

```
Terraform will perform the following actions:

  # snowflake_grant_privileges_to_account_role.grant_schema will be updated in-place
  ~ resource "snowflake_grant_privileges_to_account_role" "grant_schema" {
        id                = "\"_ROLE_TEST_MODELS\"|false|false|USAGE|OnSchemaObject|OnFuture|MODELS|InDatabase|\"SHARED\""
      ~ privileges        = [
          + "USAGE",
        ]
        # (5 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

This may be due to snowflake reporting "grant_on" as 'MODULE' in `SHOW FUTURE GRANTS IN DATABASE`.

### Steps to Reproduce

1. Copy configuration
2. Run *Terraform apply*
3. Confirm in snowflake future grant was applied. `SHOW FUTURE GRANTS TO ROLE ROLE_TEST_MODELS;`
4. Repeat *Terraform apply*

### How much impact is this issue causing?

Low

### Would you like to implement a fix?

- [ ] Yeah, I'll take it 😎

**Comment by sfc-gh-asawicki** [comment]
Hey @halcarrigan-carvana. Thanks for reaching out to us.

It seems like an undocumented Snowflake behavior. We will reproduce the behavior, contact the team responsible for grants, and decide what we will do with it. Ideally, it should be changed on the Snowflake side, but we will probably implement a workaround for the time being.

If we decide to implement the workaround, it will land in one of the next minor versions.

cc: @sfc-gh-jcieslak

**Comment by jdoldis** [comment]
@sfc-gh-asawicki on version 0.94 we are also having this issue, and when trying to remove these resources we're getting the error - 

```
Error: [grants_validations.go:173] exactly one of AccountRoleGrantPrivileges fields [AllPrivileges GlobalPrivileges AccountObjectPrivileges SchemaPrivileges SchemaObjectPrivileges] must be set
```

We haven't seen that for any other type of future grant.

Do you think that's related to this issue, or should I open a separate issue to track that?

**Comment by sfc-gh-asawicki** [comment]
cc: @sfc-gh-jcieslak

**Comment by sfc-gh-jcieslak** [comment]
Hey @jdoldis 
As far as I remember this particular error may be connected to this exact issue. Could you also post what your `snowflake_grant_privileges_to_account_role` config looks like?

**Comment by sfc-gh-jcieslak** [comment]
Ok, nothing fancy, so most likely the issues are connected to `object_type_plural` being different in SHOW GRANTS. I'll fix it as soon as I'll have some time (It should be most likely fixed in the next version of the provider).

**Comment by jdoldis** [comment]
Awesome, thanks @sfc-gh-jcieslak !

**Comment by halcarrigan-carvana** [comment]
Thank you all for the quick response on this.

**Comment by sfc-gh-jcieslak** [comment]
Hey @halcarrigan-carvana 
Sorry, for the late response, but the fix was already released as part of the 0.96.0 version. Could you validate if the grants on the model are working properly in your case? Thank You.
